frontend/src/main/components/BasicCourseSearch/CourseOverTimeBuildingsSearchForm.js

```diff
@@ -0,0 +1,84 @@
+import { useState } from "react";
+import { Form, Button, Container, Row, Col } from "react-bootstrap";
+
+import { allBuildings } from "fixtures/buildingFixtures";
+import { quarterRange } from "main/utils/quarterUtilities";
+
+import { useSystemInfo } from "main/utils/systemInfo";
+import SingleQuarterDropdown from "../Quarters/SingleQuarterDropdown";
+import SingleBuildingDropdown from "../Buildings/SingleBuildingDropdown";
+
+const CourseOverTimeBuildingsSearchForm = ({ fetchJSON }) => {
+
+  const { data: systemInfo } = useSystemInfo();
+
+  // Stryker disable OptionalChaining
+  const startQtr = systemInfo?.startQtrYYYYQ || "20211";
+  const endQtr = systemInfo?.endQtrYYYYQ || "20214";
+  // Stryker restore OptionalChaining
+
+  const quarters = quarterRange(startQtr, endQtr);
+
+  // Stryker disable all : not sure how to test/mock local storage
+  const localStartQuarter = localStorage.getItem("CourseOverTimeBuildingsSearch.StartQuarter");
+  const localEndQuarter = localStorage.getItem("CourseOverTimeBuildingsSearch.EndQuarter");
+  const localBuildingCode = localStorage.getItem("CourseOverTimeBuildingsSearch.BuildingCode")
+
+  const [startQuarter, setStartQuarter] = useState(localStartQuarter || quarters[0].yyyyq);
+  const [endQuarter, setEndQuarter] = useState(localEndQuarter || quarters[0].yyyyq);
+  const [buildingCode, setBuildingCode] = useState(localBuildingCode || {});
+
+  const rowStyle = { paddingTop: 10, paddingBottom: 10 };
```

Comment:
I think you should probably make a test for this instead of turning Stryker off for it. There's an article about how to deal with it at https://ucsb-cs156.github.io/topics/stryker/stryker_html_attribute_surviving_mutants.html.

I also have a proposed way of testing this particular case on my form for instructor search, but I'm not totally sure yet if it's the proper way to do it.

Reply:
Agreed.